cdap-explore/src/main/java/co/cask/cdap/explore/service/hive/BaseHiveExploreService.java

```diff
@@ -676,35 +671,19 @@ public QueryStatus getStatus(QueryHandle handle) throws ExploreException, Handle
     throws HiveSQLException, ExploreException, HandleNotFoundException {
     startAndWait();
 
-    try {
-      if (operationHandle.hasResultSet()) {
-        // Rowset is an interface in Hive 13, but a class in Hive 12, so we use reflection
-        // so that the compiler does not make assumption on the return type of fetchResults
-        Object rowSet = getCliService().fetchResults(operationHandle, FetchOrientation.FETCH_NEXT, size);
-        Class rowSetClass = Class.forName("org.apache.hive.service.cli.RowSet");
-        Method toTRowSetMethod = rowSetClass.getMethod("toTRowSet");
-        TRowSet tRowSet = (TRowSet) toTRowSetMethod.invoke(rowSet);
-
-        ImmutableList.Builder<QueryResult> rowsBuilder = ImmutableList.builder();
-        for (TRow tRow : tRowSet.getRows()) {
-          List<Object> cols = Lists.newArrayList();
-          for (TColumnValue tColumnValue : tRow.getColVals()) {
-            cols.add(tColumnToObject(tColumnValue));
-          }
-          rowsBuilder.add(new QueryResult(cols));
+    if (operationHandle.hasResultSet()) {
+      ImmutableList.Builder<QueryResult> rowsBuilder = ImmutableList.builder();
+      Iterable<Object[]> rowSet = getCliService().fetchResults(operationHandle, FetchOrientation.FETCH_NEXT, size);
```

Comment:
So in every cases, in all Hive versions, the result of ``fetchResults`` is always an ``Iterable<Object[]>`` ?